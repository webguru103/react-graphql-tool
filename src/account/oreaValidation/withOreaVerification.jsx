import * as React from 'react';
import type { RouterHistory, Location, Match } from 'react-router';
import OREAValidationDialog from './index';
import { withDialog } from '../../shared/dialog/withDialog';

type PropType = {
  match: Match,
  closeDialog: Function,
  createDialog: Function,
  location: Location,
  history: RouterHistory,
  props: any,
};

export function withOreaVerificationC(Component: React.Element) {
  class OreaVerificationHOC extends React.Component<PropType> {

    componentDidMount() {
      const {
        match,
        closeDialog,
        createDialog,
        location,
        history,
      } = this.props;

      const dialogLaunch = (initialDialog) => {
        createDialog({
          dialogContent: <OREAValidationDialog
            match={match}
            initialDialog={initialDialog}
            closeDialog={closeDialog}
          />,
        });
        history.push(match.path);
      };

      if (location.search !== '') {
        const params = new URLSearchParams(location.search);
        if (params.get('Response') === 'Yes') {
          dialogLaunch('verification-success');
        } else if (params.get('Response') === 'No') {
          dialogLaunch('verification-failed');
        }
      }
    }

    oreaDialog = ({ firstLogin } = { firstLogin: false }) => {
      this.props.createDialog({
        dialogContent: <OREAValidationDialog
          match={this.props.match}
          closeDialog={this.props.closeDialog}
          firstLogin={firstLogin}
        />,
        disableClickOutside: true,
      });
    };

    render() {
      return (
        <Component {...this.props} oreaDialog={this.oreaDialog} />
      );
    }

  }
  return withDialog(OreaVerificationHOC);
}

export default withOreaVerificationC;
