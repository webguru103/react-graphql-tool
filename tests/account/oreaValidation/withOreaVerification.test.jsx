/* eslint import/no-extraneous-dependencies: 0 */

import * as React from 'react';
import { func } from 'prop-types';
import 'jest-dom/extend-expect';
import { render, cleanup } from 'react-testing-library';
import withOreaVerification from '../../../src/account/oreaValidation/withOreaVerification';

const TestComponent = ({ oreaDialog }) => (
  <div data-testid="test">
    {oreaDialog ? 'true' : 'false'}
  </div>
);

TestComponent.propTypes = {
  oreaDialog: func.isRequired,
};

describe('With OREA Verification HOC', () => {
  afterEach(cleanup);
  const El = withOreaVerification(TestComponent);

  it('renders', () => {
    const { container } = render(<El classes={{}} location={{}} match={{}} />);

    expect(container).toMatchSnapshot();
  });

  it('passes in oreaDialog function', () => {
    const container = render(<El classes={{}} location={{}} match={{}} />);

    expect(container.getByTestId('test')).toHaveTextContent('true');
  });

});
