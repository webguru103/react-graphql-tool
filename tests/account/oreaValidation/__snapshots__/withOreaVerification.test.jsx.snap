// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With OREA Verification HOC renders 1`] = `
<div>
  <div
    data-testid="test"
  >
    true
  </div>
</div>
`;
